### PR TITLE
feat: [PLG-1164]: ci credit card flow

### PR DESCRIPTION
**Pull request by dhruvUpadhyay07**
#### Screenshots

Screenshot 2023-04-22 at 11 16 33 AM
Screenshot 2023-04-22 at 11 17 07 AM
Screenshot 2023-04-22 at 11 17 45 AM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by dev-1995 — APPROVED** [review]

**Review by devinder-harness — APPROVED** [review]

**Review by pranil-harness — APPROVED** [review]

File: src/modules/33-auth-settings/components/Subscription/BillingInfo/BillingInfo.tsx [hunk 1]
```diff
@@ -111,6 +128,52 @@ export const BillingInfo: React.FC<BillingInfoProp> = ({
...
+              city: data?.city,
+              country: data?.country,
+              state: data?.state
+            },
+            billingEmail: subscriptionProps.billingContactInfo.email,
+            companyName: subscriptionProps.billingContactInfo.companyName
+          }
+        })
+      } else {
+        // res = await updateSubscription({ customerId: '', data: {} })
```
Comment (avinashmadhwani02) [inline src/modules/33-auth-settings/components/Subscription/BillingInfo/BillingInfo.tsx:167]: why this code is commented?

File: src/modules/33-auth-settings/components/Subscription/BillingInfo/BillingInfo.tsx [hunk 2]
```diff
@@ -111,6 +128,52 @@ export const BillingInfo: React.FC<BillingInfoProp> = ({
...
+      }
+      setErr('')
+      return res?.data?.subscriptionId
+    } catch (error) {
+      const errorMessage = getErrorMessage(error)
+      setErr(errorMessage)
+      throw new Error(errorMessage)
```
Comment (avinashmadhwani02) [inline src/modules/33-auth-settings/components/Subscription/BillingInfo/BillingInfo.tsx:174]: Shouldn't we use toast to show this error?

File: src/modules/33-auth-settings/components/Subscription/CostCalculator/CIDeveloperCard.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  const recommendedNumber = defaultTo(recommended, Math.max(Math.ceil(usage * 1.2), currentSubscribed))
+  return (
+    <Layout.Horizontal flex={{ justifyContent: 'space-between' }} className={css.subscriptionInfo}>
+      <Layout.Horizontal>
+        <Item title={getString('authSettings.costCalculator.currentSubscribed')} value={currentPlanDescr} />
+      </Layout.Horizontal>
+      <Layout.Horizontal>
+        <Item
+          title={getString('authSettings.costCalculator.using')}
+          value={<Text font={{ weight: 'bold' }}>{usage}</Text>}
```
Comment (avinashmadhwani02) [inline src/modules/33-auth-settings/components/Subscription/CostCalculator/CIDeveloperCard.tsx:57]: Avoid using font weight, use font variation instead

File: src/modules/33-auth-settings/components/Subscription/CostCalculator/CIDeveloperCard.tsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        />
+      </Layout.Horizontal>
+      <Layout.Horizontal className={css.subscriptionInfo}>
+        <Item
+          title={getString('authSettings.recomendation')}
+          value={
+            <Text color={Color.PRIMARY_7} font={{ weight: 'bold' }}>
```
Comment (avinashmadhwani02) [inline src/modules/33-auth-settings/components/Subscription/CostCalculator/CIDeveloperCard.tsx:64]: SAA

**Review by dev-1995 — APPROVED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]
@dhruvUpadhyay07 Please sync up when you will raise the subsequent PR for this. I have some suggestions.

**Pull request merged**